autogen/main/cluster.tf.tmpl

```diff
@@ -491,7 +491,7 @@ resource "google_container_cluster" "primary" {
 
   {% if autopilot_cluster != true %}
   dynamic "dns_config" {
-    for_each = var.cluster_dns_provider == "CLOUD_DNS" ? [1] : []
+    for_each = contains(["CLOUD_DNS", "PROVIDER_UNSPECIFIED", "PLATFORM_DEFAULT"], var.cluster_dns_provider) ? [1] : []
```

Comment:
The drift appears to be when using the default `PROVIDER_UNSPECIFIED` and `DNS_SCOPE_UNSPECIFIED`, the API/Provider does not later return the `dns_config` block.  Perhaps we can be more precise by specifying when not to create this block.

```suggestion
    for_each = !(var.cluster_dns_provider == "PROVIDER_UNSPECIFIED" && var.cluster_dns_scope == "DNS_SCOPE_UNSPECIFIED" && var.cluster_dns_domain == "" ) ? [1] : []
```